mono/eglib/gstr.c

```diff
@@ -869,21 +901,22 @@ g_ascii_strncasecmp (const gchar *s1, const gchar *s2, gsize n)
 gint
 g_ascii_strcasecmp (const gchar *s1, const gchar *s2)
 {
-	const char *sp1 = s1;
-	const char *sp2 = s2;
-	
+	if (s1 == s2)
+		return 0;
+
 	g_return_val_if_fail (s1 != NULL, 0);
 	g_return_val_if_fail (s2 != NULL, 0);
-	
-	while (*sp1 != '\0') {
-		char c1 = g_ascii_tolower (*sp1++);
-		char c2 = g_ascii_tolower (*sp2++);
-		
-		if (c1 != c2)
-			return c1 - c2;
+
+	char c1;
+
+	while ((c1 = *s1)) {
```

Comment:
Style nit: are the double parens doing anything here?

Reply:
This is the idiom to quash gcc warnings about assignment within conditional expressions.
Unfortunately Visual C++ does not understand it.